### ISSUE TITLE
`pachctl port-forward` refuses to start after previous run was killed

### ISSUE THREAD

**Issue by Tryneus** [pachyderm/pachyderm#4713]
It appears that `pachctl port-forward` determines if port forwarding is already running by checking the `~/.pachyderm/config.json` file.  If the previous run was unceremoniously killed, the data may remain in the config file and a new run of `pachctl port-forward` will result in: 

```
$ pachctl port-forward
port forwarding appears to already be running for this context
```

The workaround is to delete the port-forwarder entry from the config file.

I think the usual way to solve this is using pidfiles.  We could probably do this without too much trouble using cooperatively-locked files, but there's the complication that we can port-forward any set of ports as long as it doesn't overlap with an existing port-forwarder.  We might need to support a set of port-forwarder pidfiles and refer to which ones are valid for which ports in the config file.  That or have a pidfile per forwarded port.  You figure it out.

**Comment by ysimonson** [comment]
Agreed that pidfiles would be good. Do not try to lock the config file itself, because we move the file into place on save to ensure the operation is atomic, but that surrenders the lock.

**Comment by lukemarsden** [comment]
Here is a terrible workaround in case it helps anyone else:
```
python3 -c '
import json
a = json.loads(open(".pachyderm/config.json").read())
if len(list(a.get("v2", {}).get("contexts", {}).items())) > 0 and "port_forwarders" in list(a.get("v2", {}).get("contexts", {}).items())[0][1]:
    del(list(a.get("v2", {}).get("contexts", {}).items())[0][1]["port_forwarders"])
    open(".pachyderm/config.json", "w").write(json.dumps(a, indent=4))
'
```

**Comment by reergymerej** [comment]
Thanks @lukemarsden for the path.
